libsolidity/codegen/ArrayUtils.cpp

```diff
@@ -290,7 +290,10 @@ void ArrayUtils::copyArrayToStorage(ArrayType const& _targetType, ArrayType cons
 			// stack: target_ref target_data_end source_data_pos target_data_pos_updated source_data_end
 			_context << Instruction::POP << Instruction::SWAP1 << Instruction::POP;
 			// stack: target_ref target_data_end target_data_pos_updated
-			utils.clearStorageLoop(targetBaseType);
+			if (targetBaseType->storageBytes() < 32)
+				utils.clearStorageLoop(TypeProvider::uint256());
+			else
+				utils.clearStorageLoop(targetBaseType);
```

Comment:
As an optimization, we could also use `uint256` for types up to 32 bytes.

Reply:
You mean changing `< 32` to `<=32`? I'm not sure about that when `type` is `MappingType`.

Reply:
Oh you mean, rewrite `clearStorageLoop` to clear at least 32 bytes on all types? I think this makes sense since all uses of this function has two branches, where for packed cases, the type is replaced by `uint256`.

Reply:
No, I would not make big changes to clearStorageLoop, but we can say: valueType and storageBytes <= 32